crates/core/src/value/convertible.rs

```diff
@@ -7,7 +7,41 @@ use thiserror::Error;
 pub enum Convertible {
     Number(f32),
     String(String),
-    Bool(bool),
+    Boolean(bool),
+}
+
+impl Convertible {
+    pub fn eq(&self, other: &Self, epsilon: f32) -> Result<bool, InvalidCastError> {
+        match (self, other) {
+            (Self::Number(a), Self::Number(b)) => Ok((a - b).abs() < epsilon),
+            (Self::Number(a), Self::String(b)) => {
+                let b: f32 = b.parse().map_err(InvalidCastError::from)?;
+                Ok((a - b).abs() < epsilon)
+            }
+            (Self::Number(a), Self::Boolean(b)) => {
+                let b = if *b { 1.0 } else { 0.0 };
+                Ok((a - b).abs() < epsilon)
+            }
+            (Self::String(a), Self::String(b)) => Ok(a == b),
+            (Self::String(a), Self::Number(b)) => {
```

Comment:
Implicit casting like this feels bad in Rust, but I guess we have to follow what Yarn does...

Reply:
I think I might be able to change this into something more idiomatic without changing the behavior later.
I think in Yarn the eq might be done via reference comparison, so we could cut out the casts here. I'll have to very this though.

Reply:
There, no casting anymore :D
[`80174c9` (#54)](https://github.com/Mafii/rusty-yarn-spinner/pull/54/commits/80174c9a6872fc25b267ef6535c8e204a447f43a)